File: ResourceManager/Services/Abstractions/IPermissionDataService.cs
using ResourceManager.Models;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace ResourceManager.Services.Abstractions
{
    public interface IPermissionDataService : IDataService<Permission>
    {
        Task<IList<string>> GetPermissionForRoles(IList<string> roles);
        Task<IList<Permission>> GetPermissionObjectForRoleID(int iD);
    }
}

File: ResourceManager/Services/Abstractions/IPermissionDataService.vb
Imports ResourceManager.Models
Imports System.Collections.Generic
Imports System.Threading.Tasks

Namespace ResourceManager.Services.Abstractions
    Public Interface IPermissionDataService
        Inherits IDataService(Of Permission)
        Function GetPermissionForRoles(ByVal roles As IList(Of String)) As Task(Of IList(Of String))
        Function GetPermissionObjectForRoleID(ByVal iD As Integer) As Task(Of IList(Of Permission))
    End Interface
End Namespace
